Projects/BricKuber/brickuber_lib.py

```diff
@@ -43,54 +56,73 @@ def __init__(self, robot_style, debug = False):
             self.MOTOR_GRAB_SPEED_GRAB = 400
             self.MOTOR_GRAB_SPEED_FLIP = 600
             self.MOTOR_GRAB_SPEED_REST = 400
+            
+            spin_direction = 1             
+            
         elif robot_style == "EV3":
             self.TurnTablePinion = 12
             self.TurnTableGear = 36
             
             # motor position constants (these likely need to be adjusted)
-            self.MOTOR_GRAB_POSITION_HOME      = 0
-            self.MOTOR_GRAB_POSITION_REST      = -35
-            self.MOTOR_GRAB_POSITION_FLIP_PUSH = -90
-            self.MOTOR_GRAB_POSITION_GRAB      = -130
-            self.MOTOR_GRAB_POSITION_FLIP      = -240
+            self.MOTOR_GRAB_POSITION_HOME      = -337 # -120 # 
+            self.MOTOR_GRAB_POSITION_REST      = -337 # -240 # 
+            self.MOTOR_GRAB_POSITION_FLIP_PUSH = -280 # -280 # 
+            self.MOTOR_GRAB_POSITION_GRAB      = -240 # -306 # 
+            self.MOTOR_GRAB_POSITION_FLIP      = -120 # -393 # 
             
             self.MOTOR_GRAB_SPEED_GRAB = 400
             self.MOTOR_GRAB_SPEED_FLIP = 600
             self.MOTOR_GRAB_SPEED_REST = 400
         else:
             raise ValueError("Unsupported robot style")
-        
+                
         self.BP.set_motor_limits(self.MOTOR_PORTS[self.MOTOR_TURN], 0, ((500 * self.TurnTableGear) / self.TurnTablePinion))
         
         self.home_all()
     
+    def read_encoder(self):
+        encoder = self.BP.get_motor_encoder(self.MOTOR_PORTS[self.MOTOR_GRAB])
+        debug_motor_commands("Encoder: " + str(encoder))
+    
     # find motor home positions for all motors
     def home_all(self):
         self.BP.set_motor_power(self.MOTOR_PORTS[self.MOTOR_GRAB], 15)
+        self.read_encoder()
         EncoderLast = self.BP.get_motor_encoder(self.MOTOR_PORTS[self.MOTOR_GRAB])
         time.sleep(0.1)
         EncoderNow = self.BP.get_motor_encoder(self.MOTOR_PORTS[self.MOTOR_GRAB])
         while EncoderNow != EncoderLast:
             EncoderLast = EncoderNow
             time.sleep(0.1)
             EncoderNow = self.BP.get_motor_encoder(self.MOTOR_PORTS[self.MOTOR_GRAB])
-        self.BP.offset_motor_encoder(self.MOTOR_PORTS[self.MOTOR_GRAB], (EncoderNow - 25))
```

Comment:
If this was already `self.BP.offset_motor_encoder(self.MOTOR_PORTS[self.MOTOR_GRAB], (EncoderNow - 25))` and working for the NXT style, if it works for the EV3 style as well, why the following four lines? Won't changing from `-` to `+` break NXT support?

Reply:
I have no idea why it works.  But it works.  The code should still be exactly the same still for an NXT model.

Reply:
My point is, I think you broke NXT support here. For the NXT it should be `self.BP.offset_motor_encoder(self.MOTOR_PORTS[self.MOTOR_GRAB], (EncoderNow - 25))`, and if that works with the EV3 as well, this should be reverted (no platform conditional).